query-engine/connectors/sql-query-connector/src/database/operations/update.rs

```diff
@@ -31,8 +31,9 @@ pub(crate) async fn update_one_with_selection(
         return get_single_record(conn, model, &filter, &selected_fields, &[], ctx).await;
     }
 
-    let update = build_update_and_set_query(model, args, Some(&selected_fields), ctx)
-        .so_that(build_update_one_filter(record_filter).aliased_condition_from(None, false, ctx));
+    let cond = FilterBuilder::without_top_level_joins().visit_filter(build_update_one_filter(record_filter), ctx);
```

Comment:
Like for all mutations, we don't render filters with top-level joins because we can't.